github/Branch.py

```diff
@@ -161,74 +162,61 @@ def edit_protection(
         be submitted. Take care to pass both as arguments even if only one is
         changing. Use edit_required_status_checks() to avoid this.
         """
-        assert strict is github.GithubObject.NotSet or isinstance(strict, bool), strict
-        assert contexts is github.GithubObject.NotSet or all(
+        assert isinstance(strict, _NotSetType) or isinstance(strict, bool), strict
```

Comment:
Are you going to merge https://github.com/PyGithub/PyGithub/pull/2535 first or merge this PR first? If you are going to merge 2535 first, I'll wait after it's merged to avoid extra conflect.

Reply:
Merged.